### ISSUE TITLE
Error: Multiple controllable devices found

### ISSUE THREAD

**Issue by mafyou425** [iRayanKhan/homebridge-chamberlain#21]
Hi guys, I'll be straight up here: I'm a first timer on Homebridge so forgive my ignorance and potentially stupid question. This seem different enough from the other issues that I thought it was necessary to open a new issue.

I've installed Homebridge and this homebridge-chamberlain plugin, but I keep hitting a wall with the blow error in macOS terminal:

> [7/2/2017, 12:25:40 PM] [Garage Light] Error: Multiple controllable devices found: 38349717, 38349718
    at getDeviceList.then.devices (/usr/local/lib/node_modules/homebridge-chamberlain/src/api.js:78:13)
    at process._tickCallback (internal/process/next_tick.js:109:7)
[7/2/2017, 12:25:40 PM] [Garage Door] Error: Multiple controllable devices found: 38349717, 38349718
    at getDeviceList.then.devices (/usr/local/lib/node_modules/homebridge-chamberlain/src/api.js:78:13)
    at process._tickCallback (internal/process/next_tick.js:109:7)

It repeats the error and eventually gives the following error in addition (similar one for the second device): 

> [7/2/2017, 12:48:08 PM] [Garage Door] { FetchError: request to https://myqexternal.myqdevice.com/api/v4/UserDeviceDetails/Get?filterOn=true failed, reason: getaddrinfo ENOTFOUND myqexternal.myqdevice.com myqexternal.myqdevice.com:443
    at ClientRequest.<anonymous> (/usr/local/lib/node_modules/homebridge-chamberlain/node_modules/node-fetch/index.js:133:11)
    at emitOne (events.js:96:13)
    at ClientRequest.emit (events.js:188:7)
    at TLSSocket.socketErrorListener (_http_client.js:309:9)
    at emitOne (events.js:96:13)
    at TLSSocket.emit (events.js:188:7)
    at connectErrorNT (net.js:1021:8)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickCallback (internal/process/next_tick.js:104:9)
  name: 'FetchError',
  message: 'request to https://myqexternal.myqdevice.com/api/v4/UserDeviceDetails/Get?filterOn=true failed, reason: getaddrinfo ENOTFOUND myqexternal.myqdevice.com myqexternal.myqdevice.com:443',
  type: 'system',
  errno: 'ENOTFOUND',
  code: 'ENOTFOUND' }

I only have one MyQ opener, no other garage door openers. After digging through, I've found some suggestions that maybe it's recognizing the light and opener as different devices. I've tried the "single accessory" config.json as well as the multiple accessories config. I've tried with and without "DeviceId", "requiredDeviceId", and variations of these with upper and lower case. Same error. 

Here is my current config.json with obscured personal information:
`{
	"bridge": {
		"name": "Homebridge",
		"username": "username",
		"port": 52464,
		"pin": "031-45-154"
	},

	"description": "Homebridge Accessories",

	"accessories": [{
			"accessory": "Chamberlain",
			"name": "Garage Door",
			"username": "email",
			"password": "password",
			"requiredDeviceId": "38349717"
		},
		{
			"accessory": "Chamberlain",
			"name": "Garage Light",
			"username": "email",
			"password": "password",
			"requiredDeviceId": "38349718"
		}
	]
}`

Any idea what I'm doing wrong? Would appreciate any support.

**Comment by caseywebdev** [comment]
Try just the first device, there is no MyQ light support right now

```json
"accessories": [{
  "accessory": "Chamberlain",
  "name": "Garage Door",
  "username": "email",
  "password": "password",
  "deviceId": "38349717"
}]
```

**Comment by mafyou425** [comment]
Thanks for getting back to me. I was reading in another thread that it might be recognizing the light as a controllable device so I played around with it: renamed each, swapped the names and IDs, tried just one, and then the other. No luck. Same error. 

On Jul 2, 2017, at 8:15 PM, Casey Foster <notifications@github.com> wrote:

Try just the first device, there is no MyQ light support right now

"accessories": [{
  "accessory": "Chamberlain",
  "name": "Garage Door",
  "username": "email",
  "password": "password",
  "deviceId": "38349717"
}]

**Comment by caseywebdev** [comment]
So I don't think this is actually a config issue. Can you confirm that the device running `homebridge` can actually resolve https://myqexternal.myqdevice.com/api/v4/UserDeviceDetails/Get?filterOn=true ? The error `getaddrinfo ENOTFOUND myqexternal.myqdevice.com` makes me think this is actually a network issue.

**Comment by mafyou425** [comment]
I get the following response. It seems to resolve but I might have the incorrect app ID. But, like I said, I’m totally new at this. 

{"ReturnCode":"1001","ErrorMessage":"MyQ Application ID unrecognized. (1001)","CorrelationId":"6a6e35d9-98b8-471f-a75c-339f8d354e9c"}

On Jul 2, 2017, at 9:09 PM, Casey Foster <notifications@github.com> wrote:

So I don't think this is actually a config issue. Can you confirm that the device running homebridge can actually resolve https://myqexternal.myqdevice.com/api/v4/UserDeviceDetails/Get?filterOn=true <https://myqexternal.myqdevice.com/api/v4/UserDeviceDetails/Get?filterOn=true> ? The error getaddrinfo ENOTFOUND myqexternal.myqdevice.com makes me think this is actually a network issue.

**Comment by mafyou425** [comment]
OK, I got it to work. I'm going to be completely honest, I'm not 100% sure how I fixed it, but the only recognizable change I did from what I pasted above is I removed the first accessory and changed the accessory "name" from "Garage Door" to "Controller". I'm leaning towards the name being the fix as a possible conflict with the HomeKit device name because this was the only thing I didn't try yesterday.
